### ISSUE TITLE
Conflict between packages google_sign_in (^4.0.6) & local_auth (^0.5.2+4) causing build failure

### ISSUE THREAD

**Issue by jonsamwell** [flutter/flutter#38752]
Hi,

Having the two packages google_sign_in (^4.0.6) & local_auth (^0.5.2+4) in an Androidx Flutter project causes a build error. Note that this happens on a brand new project created using ```flutter create app --androidx```.

```
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDevelopmentDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
  Program type already present: com.google.common.util.concurrent.ListenableFuture
  Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
```

If you remove either of the packages the app will build and run.

The same error happens using gradle 'com.android.tools.build:gradle:3.3.0' & 'com.android.tools.build:gradle:3.2.1'

pubspec.yaml

```
....
environment:
  sdk: ">=2.1.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  google_sign_in: ^4.0.6
  local_auth: ^0.5.2+4
....
```

Flutter doctor output

```
[√] Flutter (Channel stable, v1.7.8+hotfix.4, on Microsoft Windows [Version 10.0.18362.295], locale en-GB)
    • Flutter version 1.7.8+hotfix.4 at C:\Google\flutter
    • Framework revision 20e59316b8 (4 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    • Android SDK at C:\Users\username\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 38.2.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] VS Code (version 1.37.1)
    • VS Code at C:\Users\username\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.3.0

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by jonsamwell** [comment]
Any comment on this?

**Comment by BondarenkoStas** [comment]
@jonsamwell Please confirm was it fixed using ```google_sign_in 4.0.7```?

**Comment by VladyslavBondarenko** [comment]
@jonsamwell 
Seems to be no conflicts with using the last versions
```
google_sign_in: ^4.1.4
local_auth: ^0.6.1+3
```
Closing,
if you disagree please write in the comments and I will reopen it
thank you
